packages/webpack/src/plugin.js

```diff
@@ -339,6 +353,37 @@ class LavaMoatPlugin {
                   return
                 }
 
+                // Fixes the issue with assets being emitted to dist without a loader
+                // TODO: refactor to move random hardening of the build somewhere it's easier to track.
+                if (
+                  isAmbientAsset(module) &&
+                  module.resource.includes('node_modules') // FIXME: would be better to use canonicalName lookup and match with root
+                ) {
+                  // add a warning about removing the asset
+                  mainCompilationWarnings.push(
+                    new WebpackError(
+                      `LavaMoatPlugin: the following resource was being silently emitted to the dist and LavaMoat has prevented it: '${module.resource}'. If you want to add this resource, explicitly define a file-loader for it in your webpack configuration.`
+                    )
+                  )
+
+                  // chunkGraph.disconnectChunkAndModule(chunk, module) // can't use this, the require statement remains and errors out
+
+                  if (module.generatorOptions) {
+                    // genertorOptions was not present in testnig, but types indicate it might be there
+                    module.generatorOptions.emit = false
+                  }
+                  if (module.generator) {
+                    module.generator = Object.create(module.generator, {
```

Comment:
it's possible the object is frozen, but yeah, I would try to mutate it first

Reply:
The idea is to put a field on generator that guarantees emit is false even if generator itself is immutable. Extending it this way preserves any functionality (think methods) it could carry. There's some inconsistency between `generator` and `generatorOptions` being there, but actual generator options being in `generatpr` and `genertorOptions` being empty in practice. Looks like a leftover from previous version maybe. 
I arrived at this with trial and error, seems like the most compatible way to override `emit`